### DISCUSSION TITLE
Stream download with Laravel vapor

### DISCUSSION THREAD

**Discussion by chrisrhymes** [spatie/simple-excel#149] [Q&A]
I am using Laravel vapor to host my Laravel app and when I use the `SimpleExcelWriter::streamDownload('my-file.xlsx')` vapor returns `{"message": "Internal server error"}`. If I change the extension to .csv instead of Excel then it seems to work as expected. 

The [vapor docs](https://docs.vapor.build/1.0/projects/development.html#binary-responses) mention that you need to add an additional header to the response for binary responses, is there a way of doing this with the `streamDownload()->toBrowser()` method?

**Comment by od3n** [comment]
Have you found a solution for the issue @chrisrhymes?

**Comment by Tickthokk** [comment]
`header('X-Vapor-Base64-Encode: True');` before my `::streamDownload` call was my solution, which might be obvious, but it would be great to have an official way to append a response header.
